### PR TITLE
Add credit method to Komoju recurring payment methods

### PR DESCRIPTION
**Pull request by shioyama**

### CODE REVIEW

**Comment by shioyama** [comment]
Together with degica/active_merchant#23, this will allow refunds for komoju recurring payment methods.

**Comment by shioyama** [comment]
Actually degica/active_merchant#23 added void, which we're not using here... so not required. Using `provider.refund`, I don't think we actually need `void`.

File: app/models/spree/gateway/komoju_recurring.rb [hunk 1]
```diff
@@ -5,10 +5,23 @@ def provider_class
...
+        true
+      end
+
       def auto_capture?
         true
       end
 
+      def credit(amount, source, authorization, options = {})
+        amount = cents_to_dollar(amount) if options[:currency] == "JPY"
+        provider.refund(amount, authorization, options)
```
Comment (shioyama) [inline app/models/spree/gateway/komoju_recurring.rb]: Important thing here is that Spree will send the `source` as the second parameter if the payment method has profiles supported, which both the initial and additional payment methods do. However, the provider (Komoju AM gateway) does not accept this format of arguments, so we just need to ignore the source.

See [here](https://github.com/degica/liquidus/blob/51e573ee78383213e6abef6c0291ca1e00c4d9bd/core/app/models/spree/payment/processing.rb#L42).

**Comment by degicat** [comment]
@miyazawadegica Please review this.

**Comment by shioyama** [comment]
Added a few commits to fix unrelated spec warnings, etc.

**Pull request merged**